### ISSUE TITLE
Squiz.Commenting.FunctionCommentThrowTag throw $e error

### ISSUE THREAD

**Issue by ofbeaton** [squizlabs/PHP_CodeSniffer#650]
`Squiz.Commenting.FunctionCommentThrowTag.WrongNumber` is reporting errors when you have code like the following:

``` php
/**
 * @throws RuntimeException Here just to have multiple exception.
 * @throws MyDbException When databases burn.
 */
function myMethod() {
  try {
    somecode();
  } catch (MyDbException $e) {
    if (doacheckone($e) === true) {
      throw $e;
    }
  }
  throw new \RuntimeException('Just another exception for example sake');
}
```

It should really be detecting that the `throw $e` is valid.

**Comment by gsherwood** [comment]
This is a known issue, but it's not something that the sniff is ever going to check for. If you want something much more specific, you'll need to write a custom sniff that tries to follow the code paths and see if the variable represents a specific exception.

**Comment by ofbeaton** [comment]
Sorry, I don't think I was clear enough in my post. I'm not talking about `Squiz.Commenting.FunctionCommentThrowTag.Missing` which would try trickery to see if the `@throw` matched the exception class.

I was talking about `Squiz.Commenting.FunctionCommentThrowTag.WrongNumber` which is really just asking "how many `throw` tokens are there in the method?". Unless I'm the one misunderstanding and it's doing some magic to figure out duplicates and eliminate them based on the exception class? If it is just counting how many `throw`, then `throw $e` should count. 

Even if you're doing some trickery to detect duplicates, couldn't it just count `throw $e` as one? Right now it stops me from documenting it and I had to disable the warning.

I'm fine implementing this in my own sniff however if it's not desired in the core.

**Comment by gsherwood** [comment]
Sorry, my bad. Thanks for explaining it to me again.

**Comment by virgofx** [comment]
Any progress on this ?

**Comment by egyptianbman** [comment]
Also just hit this and noticed this has been in place since 2015 which is unfortunate.  Without the second `@throws`, I get a `Squiz.Commenting.FunctionCommentThrowTag.Missing`. If I then add it, I get a `Squiz.Commenting.FunctionCommentThrowTag.WrongNumber`. This has led me to have to disable automated phpcs testing until it's resolved.

**Comment by gsherwood** [comment]
> This has led me to have to disable automated phpcs testing until it's resolved.

Or just remove this sniff from your standard.

**Comment by webimpress** [comment]
@egyptianbman can you provide your code which fails? What version of PHP_CodeSniffer are you using?
I've tested on last version (dev-master) and I can't see any issue with code provided by @ofbeaton. It works fine for me.

**Comment by egyptianbman** [comment]
I've tried wrapping the tray catch with ignore start and end comments as described in https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file, to no avail.

We actually have this installed in dev-web with version "*" requested in composer.json. I verified that the issue persists with a `composer.phar update squizlabs/php_codesniffer`

This is in a method with two blocks that `throw $e`. The first, is something like the following:

```php
            try {
                $this->someService->someFunc(
                    $param1,
                    $param2,
                    $param3/100,
                    true,
                );
            } catch (\Exception $e) {
                throw $e;
            }
```

In another block, I'm seeing:
```php
                try {
                    $someObject->someParam = $someObject->param2 - ($anotherObject->anotherParam - $param3);
                    $someObject->save();
                } catch (\Exception $e) {
                    $this->logger->addInfo(
                        "$logPrefix something failed" .
                        "some information",
                        ['error' => $e->getMessage()]
                    );
                    throw $e;
                }
```

**Comment by gsherwood** [comment]
This was actually fixed in version 2.9.0 and 3.0.0 by PR #946. @egyptianbman What version are you running?

**Comment by gsherwood** [comment]
Closing as I believe this is fixed.
